### PR TITLE
Fixed a bug when a COF layer details become invisible on combo layer selection

### PR DESCRIPTION
**Pull request by 1Dz87**

### CODE REVIEW

**Review by gravestench — DISMISSED** [review]

**Review by gucio321 — APPROVED** [review]

**Comment by gucio321** [comment]
@1Dz87 conflicts

File: hswidget/cofwidget/widget.go [hunk 1]
```diff
@@ -71,6 +71,51 @@ func (p *widget) Build() {
...
+		},
+		newLayerFields: &newLayerFields{
+			selectable: true,
+			drawEffect: int32(d2enum.DrawEffectNone),
+		},
+	}
+
+	if len(p.cof.CofLayers) > 0 {
+		state.viewerState.layer = &p.cof.CofLayers[0]
+	}
```
Comment (gucio321) [inline hswidget/cofwidget/widget.go]: please move this into `hswidget/cofwidget/state.go

**Review by gucio321 — CHANGES_REQUESTED** [review]
in one of my PRs I moved state management stuff into `hswidget/cofwidget/state.go`. Please make your changes there.
(build errors)

File: hswidget/cofwidget/widget.go (hunk 1 above)
Comment (1Dz87) [reply hswidget/cofwidget/widget.go]: There is no need of doing this. You have already moved it.

Comment (gucio321) [reply hswidget/cofwidget/widget.go]: @1Dz87  but your change are still necessary in state.go

Comment (1Dz87) [reply hswidget/cofwidget/widget.go]: @gucio321 It's there... You moved it in lines 224 - 226 of state.go

Comment (gucio321) [reply hswidget/cofwidget/widget.go]: @1Dz87 I meant, that I moved the getState method into state.go, but the following code, you added here, is still necessary in state.go (below this line: https://github.com/OpenDiablo2/HellSpawner/blob/5f782226b49ea120a74f909dfc2b224db31dc843/hswidget/cofwidget/state.go#L199):
```go
         if len(p.cof.CofLayers) > 0 {
		state.viewerState.layer = &p.cof.CofLayers[0]
	}
```

Comment (1Dz87) [reply hswidget/cofwidget/widget.go]: Oh, sorry! A bit misunderstood( Now done.

**Review by gucio321 — APPROVED** [review]

**Review by gravestench — APPROVED** [review]

**Pull request merged**